cloud/src/autogluon/cloud/utils/iam.py

```diff
@@ -0,0 +1,295 @@
+TRUST_RELATIONSHIP_FILE_NAME = 'ag_cloud_sagemaker_trust_relationship.json'
+IAM_POLICY_FILE_NAME = 'ag_cloud_sagemaker_iam_policy.json'
+
+TRUST_RELATIONSHIP_ACCOUNT_PLACE_HOLDER = 'ACCOUNT'
+POLICY_ACCOUNT_PLACE_HOLDER = 'ACCOUNT'
+POLICY_BUCKET_PLACE_HOLDER = 'CLOUD_BUCKET'
+
+SAGEMAKER_TRUST_RELATIONSHIP = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "sagemaker.amazonaws.com",
+                "AWS": f"arn:aws:iam::{POLICY_ACCOUNT_PLACE_HOLDER}:root"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+
+SAGEMAKER_CLOUD_POLICY_NAME = 'AutoGluonSageMakerCloudPredictor'
+SAGEMAKER_CLOUD_POLICY_DESCRIPTION = 'AutoGluon CloudPredictor with SageMaker Backend Required Policy'
+
+SAGEMAKER_CLOUD_POLICY = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "SageMaker",
+            "Effect": "Allow",
+            "Action": [
+                "sagemaker:DescribeEndpoint",
+                "sagemaker:DescribeEndpointConfig",
+                "sagemaker:DescribeModel",
+                "sagemaker:DescribeTrainingJob",
+                "sagemaker:DescribeTransformJob",
+                "sagemaker:CreateArtifact",
+                "sagemaker:CreateEndpoint",
+                "sagemaker:CreateEndpointConfig",
+                "sagemaker:CreateModel",
+                "sagemaker:CreateTrainingJob",
+                "sagemaker:CreateTransformJob",
+                "sagemaker:DeleteEndpoint",
+                "sagemaker:DeleteEndpointConfig",
+                "sagemaker:DeleteModel",
+                "sagemaker:UpdateArtifact",
+                "sagemaker:UpdateEndpoint",
+                "sagemaker:InvokeEndpoint"
+            ],
+            "Resource": [
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:artifact/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:transform-job/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:endpoint/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:training-job/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:model/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:endpoint-config/*"
+            ]
```

Comment:
[minor] for security sensitive customers, can we make sure all the SM resources created with prefix `ag` so that we can further constraint the resource access required? Some customers might be reluctant to allow us to access other resources which might not be related to AG at all

Reply:
I think this is a good idea. This would require me look into how to add the prefix assuming it's doable. We can have a follow up PR that modify both the source code and further scope down the policy

Reply:
+1 on follow-up PR to add prefixes